roles/ci_local_storage/tasks/main.yml

```diff
@@ -0,0 +1,87 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Create role needed directories
+  ansible.builtin.file:
+    path: "{{ cifmw_cls_manifests_dir }}"
+    state: directory
+
+- name: Create the cifmw_cls_namespace namespace"
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    name: "{{ cifmw_cls_namespace }}"
+    kind: Namespace
+    state: present
+
+- name: Save storage manifests as artifacts
+  ansible.builtin.copy:
+    dest: "{{ cifmw_cls_manifests_dir }}/storage-class.yaml"
+    content: "{{ cifmw_cls_storage_manifest | to_nice_yaml }}"
+
+- name: Apply the storage class manifests
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig  }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit)  }}"
+    state: present
+    src: "{{ cifmw_cls_manifests_dir }}/storage-class.yaml"
+
+- name: Get the worker nodes
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    kind: Node
+    label_selectors:
+      - node-role.kubernetes.io/worker
+  register: _node_info
+
+- name: Get all the worker nodes
+  ansible.builtin.set_fact:
+    cifmw_cls_nodes: >-
+      {{ _node_info.resources |
+          selectattr("metadata.name", "defined") |
+          map(attribute="metadata.name")
+      }}
+
+- name: Generate node_storage script
+  ansible.builtin.template:
+    src: node_storage.sh.j2
+    dest: "{{ cifmw_cls_manifests_dir }}/node_storage.sh"
+    mode: "a+x"
+
+- name: Run node_storage script to create directories on worker node
```

Comment:
Do we really need a bash script for running an `oc` command in a loop?

Reply:
yeah, I'd switch to an `ansible.builtin.shell` task with the `oc` loop. I guess we can also get rid of the messages (since we don't register the execution of the task) and just create or remove folders.

Reply:
Make sense, Done!